catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/CatalogFrameworkImpl.java

```diff
@@ -1522,6 +1525,85 @@ public UpdateResponse update(UpdateRequest updateRequest)
         return updateResponse;
     }
 
+    private DeleteRequest rewriteQuery(DeleteRequest deleteRequest)
+            throws UnsupportedQueryException {
+        if (Metacard.ID.equals(deleteRequest.getAttributeName())) {
+            return deleteRequest;
+        }
+
+        Filter filter = frameworkProperties.getFilterBuilder()
+                .anyOf(deleteRequest.getAttributeValues()
+                        .stream()
+                        .map(e -> frameworkProperties.getFilterBuilder()
+                                .attribute(deleteRequest.getAttributeName())
+                                .is()
+                                .equalTo()
+                                .text((String) e))
+                        .collect(Collectors.toList()));
+        SourceResponse response = catalog.query(new QueryRequestImpl(new QueryImpl(filter), false));
+        List<Entry<Serializable, Metacard>> newUpdateList = response.getResults()
```

Comment:
Even if order matters, you could do a `Collectors.toMap` with a `LinkedHashMap::new`.